src/gpu/intel/ocl/stream.cpp

```diff
@@ -136,6 +136,81 @@ status_t stream_t::barrier() {
     return impl()->barrier();
 }
 
+status_t stream_t::init_async_tracking(std::string &vinfo, double *start_ms) {
+    stream_timing_data_->start_ms = get_msec();
```

Comment:
After taking a second look - it seems we need some changes to ensure it's thread-safe. It's not safe to have `stream_timing_data_` stored inside the stream object as there may be multiple concurrent submissions to the same stream (which is a valid scenario).

There is an option to use a mutex specifically with `ONEDNN_VERBOSE` mode but I would try to explore less intrusive options first - to avoid deviations with verbose mode and just to follow the common design.

Please take a look at `xpu::{sycl,ocl}::context_t`:

https://github.com/uxlfoundation/oneDNN/blob/608fc3ae0b6ee5d53d2a60dc97263f322a020838/src/xpu/ocl/stream_impl.hpp#L98

It's used to track events, and TLS helps to avoid issues with concurrent execution. Can we move the "timing data" structure to the context?